franklin/builder/models.py

```diff
@@ -39,34 +154,32 @@ class Site(models.Model):
         (SUCCESS, 'Deploy Succeeded'),
         (FAILED, 'Deploy Failed')
     )
-
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    owner = models.CharField(max_length=100, default='')
-    owner_id = models.PositiveIntegerField(blank=True, null=True)
-    repo_name = models.CharField(max_length=100)
-    repo_name_id = models.PositiveIntegerField(blank=True, null=True)
-    git_hash = models.CharField(max_length=40)
-    url = models.CharField(max_length=100, default='', db_index=True)
-    path = models.CharField(max_length=100)
-    oauth_token = models.CharField(max_length=255, default='')
+    
+    site = models.ForeignKey(Site, related_name='environments')
+    name = models.CharField(max_length=100, default='')
+    description = models.TextField(max_length=20480, default='', blank=True)
+    deploy_type = models.CharField(
+        max_length=3, choices=DEPLOY_CHOICES, default=BRANCH)
+    branch = models.CharField(max_length=100, default='master')
+    tag_regex = models.CharField(max_length=100, blank=True)
+    url = models.CharField(
+        max_length=100, default='', db_index=True, blank=True)
+    current_deploy = models.ForeignKey(
+        Build, related_name='deployments', null=True, blank=True)
+    past_builds = models.ManyToManyField(
+        Build, related_name='environments', blank=True)
     status = models.CharField(
         max_length=3, choices=STATUS_CHOICES, default=REGISTERED)
-
-    def is_deployable_event(self, github_event):
-        # TODO - add logic here to compare data from the github webhook event
-        # with what we already have in the DB about acceptable deploy events.
-        # e.g. Only if branch 'release' and has a tag formatted 'v ##.##.##'
-        return True
-
+    
     def build(self):
         url = os.environ['BUILDER_URL']
         headers = {'content-type': 'application/json'}
         body = {
-                    "github_token": self.oauth_token,
-                    "git_hash": self.git_hash,
-                    "repo_owner": self.owner,
-                    "path": self.path,
-                    "repo_name": self.repo_name
+                    "github_token": self.site.deploy_key,
+                    "git_hash": self.current_deploy.git_hash,
+                    "repo_owner": self.site.owner.name,
+                    "path": self.current_deploy.path,
+                    "repo_name": self.site.name
```

Comment:
per chat. yes....will need to included something here to represent branch/tag (ref?)
builder logic:
if git_hash is present, checkout to the has after cloning. Otherwise clone is good enough.